### DISCUSSION TITLE
Save TaskBar ToolBar settings?

### DISCUSSION THREAD

**Discussion by PGomersall** [valinet/ExplorerPatcher#395] [General]
Valinet,
Like you have done for tray icons, would it be possible to save ToolBar settings and their icon positions if they have them?
I tried to manually export and restore [HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Explorer\Streams\Desktop] "TaskbarWinXP" but that did not seem to work

**Comment by valinet** [comment]
The main thing to do is to figure out where in the registry this is stored. Then, the approach is to redirect that reg entry in explorer.exe from the actual location to some other location, so that explorer.exe can save the settings there in peace and the other new actor in Windows 11 that resets this at every reload resets the original, but now unused registry key. That’s how it is done for the notification area icons. Maybe the approach would be similar here as well.

**Comment by PGomersall** [comment]
Maybe something you can look for "in your free time"?
I did do a compare on an export of the Explorer key in the registry while having the standard address bar loaded and after it is removed and I did see changes still in the "TaskbarWinXP" value. Maybe it is still using that setting but I have not been using (backup\restore) it correctly with ExplorerPatcher - not sure when it needs to restored into the registry - i.e. while Explorer is unloaded maybe?

**Comment by valinet** [comment]
Mitigated this issue in the latest release. Beware that it will reset your toolbar layouts one more time when you install this update, but then it should be fine. To avoid this, close Explorer gracefully (`Ctrl`+`Shift`+ right click on the taskbar, choose "Exit Explorer"); then manually export:

```
Computer\HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Explorer\Streams\Desktop\TaskbarWinXP
```

Open the generated regfile, replace `TaskbarWinXP` with `TaskbarWinEP`, save and import that regfile. Start Explorer and you should see your toolbars. Then switch to Windows 11 taskbar and back to Windows 10 taskbars and the layouts should be preserved.

Thanks for letting me know it's called `TaskbarWinXP`, it was a good starting point. The way these kind of fixes work is that I redirect the registry accesses for this keys to some other location, so that `explorer.exe` read/writes from there just as usual, while whatever new component is intentionally resetting the regentries when enabling the Windows 11 taskbar writes to the original location still, which is not used anymore.

**Comment by PGomersall** [comment]
Working great Thanks Valinet.
